intake/source/discovery.py

```diff
@@ -97,6 +98,22 @@ def enabled_plugins(self):
         out = {k: v for k, v in out.items() if k not in self.disabled()}
         return out
 
+    def openers(self):
+        """From the current state of ``registry``, create open_* functions"""
+
+        openers = set()
+        for name in self.enabled_plugins():
+            func_name = "open_" + name
+            if not func_name.isidentifier():
+                # primitive name normalization
+                func_name = re.sub("[-=~^&|@+]", "_", func_name)
+            if func_name.isidentifier():
+                # stash name for dir() and later fetch
```

Comment:
Can you check the overhead from running this function? We are not potentially running it many times, like every time the user does `intake.<tab>`.

Reply:
That's a good catch. I didn't realize `intake.__getattr__` is called so many times (~18) on a single `intake.<tab>`. I'll move this to a cache!

Reply:
18 sounds excessive! Maybe would be fewer in ipython and even more in raw python console. Still, if 18x the running time is still <1ms (should be), I wouldn't change anything.